Make it possible to play other quests

eauchat opened the pull request:
This is a proposal of some modifications to the code to allow to play other quests.
Additionally, it includes the following additional quests:
 - finding USA states (which was already partially implemented, at least the data files where already present),
 - Chinese provinces (new :)),
 - French departments (new :)).

This PR should already be usable as is. But it's not meant to be a finalized project. It's more like a first step towards making geoquest versatile enough to support many types of different quests.

Hope you find it useful :)

woutdp commented:
Just checked it out, super cool and thank you for putting in the effort!

I'm considering merging this as is, as it's better to have this feature than not. It doesn't really affect the main game so some players might enjoy it even with some of the bugs.

Some notes on it, not sure if you want to tackle any of them before I merge this:
- Circles are indeed an issue, I can take a look at this myself
- The colors of the adjacent geographic locations should be a different color than the adjacent geographic location. Basically this: https://en.wikipedia.org/wiki/Four_color_theorem It's kind of annoying to manually do it but that's something I had to do when doing the worldmap, not sure if you're up for it
- USA is so big with its territories, it's kind of annoying sometimes because the arrow points in a wrong direction. Not sure how to solve this though and probably something we have to live with

eauchat commented:
> I'm considering merging this as is, as it's better to have this feature than not. It doesn't really affect the main game so some players might enjoy it even with some of the bugs.

Sounds good to me yes :)
Also I just fixed annoying bugs that where created with the new achievements I added, hope it works well now.

> Some notes on it, not sure if you want to tackle any of them before I merge this:
>  - Circles are indeed an issue, I can take a look at this myself

If you want to have a look into it that'd be great.

>  - The colors of the adjacent geographic locations should be a different color than the adjacent geographic location. Basically this: https://en.wikipedia.org/wiki/Four_color_theorem It's kind of annoying to manually do it but that's something I had to do when doing the worldmap, not sure if you're up for it

Thanks for the link, made me learn something :)
Colors should now be fixed by latest commits.

>  - USA is so big with its territories, it's kind of annoying sometimes because the arrow points in a wrong direction. Not sure how to solve this though and probably something we have to live with

Yes, before posting the PR I was also looking at solutions, but couldn't figure out a way. I guess the only way would be to remove far-away territories. But I was actually thinking of doing the opposite and adding the missing overseas departments in the french map (still planning to do it later on). To me it's good to have all territories included, and the thing about the arrow pointing to the wrong direction also happens in the world map when one is guessing really wrong, I don't know where it comes from, but it's just a tiny bug.

Also I'll eventually add a section to the `Contributing` section of the readme explaining how to propose new maps. But don't know when, so I'd say it can be merged first and further additions can come later.

woutdp commented:
Merged and deployed, thank you for the contribution!

eauchat commented:
Great!
I'm also working on a couple of other maps and will post a new PR when they're ready :)
